src/index.ts

```diff
@@ -1,74 +1,100 @@
 import { File, WorkFlowContext, Plugin } from "fuse-box";
-import * as path from "path";
-import * as fs from "fs";
-import { spawn } from "child_process";
+import { ChildProcess } from "child_process";
+import { resolve } from "path";
+import { readFile } from "fs";
 import { tmpName } from "tmp";
+import * as spawn from "cross-spawn";
 
-const tmp = () => new Promise((resolve, reject) => tmpName((err, path) => err ? reject(err) : resolve(path)));
+const tmp = () =>
+  new Promise((resolve, reject) =>
+    tmpName((err, path) => (err ? reject(err) : resolve(path)))
+  );
 
 export interface ElmPluginOptions {
-  warn?: boolean,
-  debug?: boolean
+  warn?: boolean;
+  debug?: boolean;
 }
 
 export class ElmPluginClass implements Plugin {
-
   // Match Elm files
   public test: RegExp = /\.elm$/;
 
   public options: ElmPluginOptions;
 
   constructor(options: ElmPluginOptions = {}) {
-    this.options = Object.assign({}, options);
+    this.options = { ...options };
   }
 
-  public init(context: WorkFlowContext) {
+  public init(context: WorkFlowContext): void {
     context.allowExtension(".elm");
   }
 
-  public transform(file: File) {
-    file.loadContents();
+  public getElmMakePath(): string {
+    try {
+      return resolve("node_modules/.bin/elm-make");
+    } catch (_) {}
 
-    // Get the path to elm-make
-    const elmMakePath = getElmMakePath();
-
-    return tmp()
-      .then(tmpFilename => tmpFilename + ".js")
-      .then(tmpFilename => new Promise((resolve, reject) => {
-        // Construct the arguments for elm-make
-        const args = [
-          "--yes",
-          "--output",
-          tmpFilename,
-          this.options.warn ? "--warn" : null,
-          this.options.debug ? "--debug" : null,
-          file.absPath
-        ].filter(x => x !== null);
-
-        const proc = spawn(elmMakePath, args, { stdio: "inherit" });
-
-        proc.on("close", code => {
-          if (code === 0) {
-            fs.readFile(tmpFilename, (err, data) => {
-              if (err) reject(err);
-              file.contents = data.toString();
-              resolve(file);
-            });
-          } else {
-            reject("Failed to compile Elm");
-          }
-        });
-      }));
+    return "elm-make";
   }
 
-}
+  public handleError(elmMakePath: string, err: NodeJS.ErrnoException): void {
+    if (err.code === "ENOENT") {
+      console.error(
```

Comment:
So, after some testing it seems like you don't need to call `console.error` - instead if you reject the main promise then the error message still gets displayed, and it will integrate better with fuse-box (for example, I think it will get displayed in the browser when using HMR).